### ISSUE TITLE
Content ideas

### ISSUE THREAD

**Issue by bmcfee** [librosa/blog#1]
- [x] explaining resample-on-load
- [x] introducing the block streaming interface
- [ ] how to do feature inversion
- [ ] how to make cqt's go fast
- [ ] how to use caching effectively
- [ ] why the name? / history of the project

What else should we post about?

**Comment by bmcfee** [comment]
@lostanlen I also drafted up a post on block streaming, care to give it a looksy before I publish?

**Comment by lostanlen** [comment]
* i would love to see a post about `waveplot` and how it keeps a low memory usage. The email to John Davis from a few days ago (copy-pasted with permission on this StackOverflow board) could be used as a starting point: https://stackoverflow.com/questions/57260065/why-does-librosa-plot-differ-from-matplotlib-and-audacity

* as i wrote earlier on the Slack channel, `librosa.util.valid_audio` is one of my most cherished yet under-appreciated librosa features. I would be in favor of giving it more publicity :) we could write a short post about that.

**Comment by bmcfee** [comment]
Maybe also a plot on all the bells and whistles of `specshow`, compared to (say) imshow.

**Comment by lostanlen** [comment]
Yes, `specshow` is a major feature in librosa, and some uses are not trivial. For example:
* Explaining how the color map gets assigned (`magma` vs diverging) + a cautionary word against `jet`?
* How to make sure that the axis ticks show physical units, not frame indices (one of my biggest pet peeve as a reviewer!).
* How to make sure that the contrast is right.
* How to have several spectrograms share the same colormap and the same axes
* How to make the y-axis in CQT specshow be scientific pitch
* How to annotate a spectrogram with `piptrack`
* How to annotate a spectrogram with `beat_track`

This covers probably more than one post, but it's all very relevant IMO.

**Comment by bmcfee** [comment]
SGTM in general! I'm not sure what exactly you have in mind for `piptrack`, can you elaborate?

**Comment by lostanlen** [comment]
What i mean is to have a pitch curve (or spectral centroid, or whatever frequency-valued signal) overlayed on top of the CQT

**Comment by bmcfee** [comment]
Post ideas to accompany the 0.7.1 release:

- [ ] framing
- [ ] feature inversion (incl cqt griffin-lim)
- [ ] replicating HTK features
- [ ] pitch tracking and plotting?

**Comment by lostanlen** [comment]
`replicating HTK features`

Yes. I think that the big take-home message here is that there is no such thing as "_the_ MFCC" of a signal. It all depends on what well-established implementation we're trying to replicate, which ultimately falls back on which psychometric study we're relying on for the mel scale.
This (under-cited) paper about "fitting the mel scale" was a big eye-opener for me in this regard:
https://ieeexplore.ieee.org/document/758101

Pitch tracking and plotting: yes! will submit a WIP PR tonight.

**Comment by bmcfee** [comment]
We should probably do a comprehensive post about all the various efficiency improvements in the 0.10 release.
